framework/src/postprocessors/ElementalVariableValue.C

```diff
@@ -39,6 +38,16 @@ ElementalVariableValue::ElementalVariableValue(const InputParameters & parameter
   if (_mesh.getMesh().allow_renumbering())
     mooseError("ElementalVariableValue should only be used when node renumbering is disabled.");
 }
+
+void
+ElementalVariableValue::initialSetup()
+{
+  // Do this in initialSetup instead of the constructor because elements may be deleted after this
+  // object is constructed
+  _element = _mesh.queryElemPtr(getParam<unsigned int>("elementid"));
```

Comment:
this seems problematic

shouldnt we stop deleting elements after the mesh is done? because otherwise the ids after --mesh-only wont match the ids during the simulation

Reply:
We have had late geometric ghosting since the inception of relationship managers. This is to allow algebraic objects to give us more information about what needs to be ghosted after their construction. For instance we do this with FV kernels a lot; we inspect some information in the FV object constructor and then decide whether to expand the number of ghost layers. So the concept has always been around. Just some objects have been able to get away with not respecting that possibility until now

Reply:
> because otherwise the ids after --mesh-only wont match the ids during the simulation

I don't really see a problem if the element ids present after a `--mesh-only` are a superset of what's available during the simulation. If there is a problem, then that means somebody hasn't properly advertised their ghosting need

Reply:
I guess the problem is not new. The ids can be re-numbered between --mesh-only and a regular run

Reply:
People should just never use Ids I guess. It s too risky with the potential renumbering.

Reply:
Yes relying on ids was a bad idea even before `DistributedMesh` existed